.docker/Dockerfile

```
@@ -10,6 +10,10 @@ COPY /docker_run_git.sh /tmp/
 RUN mkdir -p /var/www/.npm
 RUN chown -R www-data:www-data /var/www/.npm
 
+# These two directories are docker mounted volumes
+RUN chown -R www-data:www-data /var/www/html/vendor
+RUN chown -R www-data:www-data /var/www/html/var
```

`/var/www` is a $HOME directory of www-data user so this should be enough for all paths, including `/var/www/.npm`, `/var/www/.composer` or whatever:
```suggestion
RUN chown -R www-data:www-data /var/www
```

---

I will have to respectfully disagree.

I would argue the host user is the owner of these files, not the container, and this will introduce unexpected behavior. If the user happens to have any other files with his directory this will overwrite its permissions every time a image is built. A system should use the permissions it requires, and nothing more.

The Dockerfile creates the user:group www-data:www-data with UID and GID 1000, this is hardcoded, and **its a failed attempt to match the host user ids**. Accidentally in Ubuntu UID and GID 1000 corresponds, by default, with the first non privileged user of the system in a fresh install.

The Dockerfile in this branch then proceeds to set some directories permissions with this UID:GID and run the apache process with it. This works but only accidentally, and the user is able to make forks, changes and commits **in the host** because www-data **in the container** has the same ids as the user in the host.

If the user has other UID:GID, and this can happen for several reasons, the proposed change will set the directories of the repository in the docker host to a different UID:GID than what the user has in the host. If the user happens to not have the same ids, heck, even if the system doesn't have the IDs defined then those files/directories would not be accessible to the host user anymore (Linux will happily let you set ownership to non-existent ids).

I see that `develop` branch [attempts to fix this](https://github.com/PrestaShop/PrestaShop/blob/develop/docker-compose.yml#L24) by passing the user and group id the Dockerfile will later use, and providing `1000` as a reasonable default value. Unfortunately it does not instruct the user how to use it in the README.

IMO this behavior needs to be incorporated in the Dockerfile of this branch AND documented in the README. 
I personally would prefer a sh-compatible script that sets the UID and GID vars using the standard GNU `id` command before invoking docker-compose. This should work in any Linux/GNU distro (maybe even Windows with WSL). 

```$ USER_ID=`id -u` GROUP_ID=`id -g` docker-compose up```

I can provide these changes if required, but as the minimum fix that only works on specific cases, I'd use my initial changes.

---

I have my own modified config, where I setup user in both Dockerfile and docker-compose.yml. and it works like this: any file created in the host is in my case owned by `sharak(1000)` and the same file mounted in the container's volume is owned by `www-data(1000)`. Files created inside container are owned by `www-data(1000)` and the same file visible on my host is owned by `sharak(1000)`. That's how it's supposed to work. Isn't it?

---

Yes, the important bit is that uid and gids need to match, current process does not guarantee this nor provides a way to achieve it, its just hardcoded.

---

In your case, yes. The important bit is that uid and gids need to match, current process does not guarantee this nor provides a way to achieve it, its just hardcoded.

---

I'm aware. Couldn't make it work though so I started using `.env` file where I can setup username, userid and groupid manually. At least I don't need to change docker setup.

Anyway, on WSL I'm the owner of my $HOME and everything inside. I make the same with www-data and container's `/var/www` and I've never had any problems.

---

@mpaolino 
Actually, if the changes done on develop (#25949) works for you,it's better to backport it on this branch.
and yead to use it like this `USER_ID=$(id -u) GROUP_ID=$(id -g) docker-compose up --build --force-recreate`

---

Yes, like I said that is the best way of doing it right now. The problem is that is a new functionality that changes a user interface (the deploy system) and might break other peoples systems, which means this probably will be required to be released as a new functionality, not a a bug fix. 

I have gone thru other bug reporting and PRs and frankly I can't imagine the level of bureaucracy, explaining and time effort required for a small change like that. 
I can't really afford the time, thanks anyway but I'm done contributing for this project, please consider closing this and provide the definitive fix as new PR. Thank you.

---

> Yes, like I said that is the best way of doing it right now. The problem is that is a new functionality that changes a user interface (the deploy system) and might break other peoples systems, which means this probably will be required to be released as a new functionality, not a a bug fix.

If you feel that's the safest way to do it, it will be tested and merged soon (as it's already approuved).

> I have gone thru other bug reporting and PRs and frankly I can't imagine the level of bureaucracy, explaining and time effort required for a small change like that. I can't really afford the time, thanks anyways but I'm done contributing to this project, please consider closing this and provide the definitive fix as new PR. Thank you, and sorry.

Sorry that you feel this way, we will try to backport the fix and test it (if possible).
And thank you for your contribution.